pydantic/error_wrappers.py

```diff
@@ -105,13 +105,21 @@ def flatten_errors(
 
 @lru_cache()
 def get_exc_type(cls: Type[Exception]) -> str:
-
-    base_name = 'type_error' if issubclass(cls, TypeError) else 'value_error'
-    if cls in (TypeError, ValueError):
-        # just TypeError or ValueError, no extra code
-        return base_name
-
-    # if it's not a TypeError or ValueError, we just take the lowercase of the exception name
+    type_names = {
+        TypeError: 'type_error',
+        ValueError: 'value_error',
+        AssertionError: 'assertion_error'
+    }
+
+    try:
```

Comment:
This seems closer to the original logic, and doesn't require the modification afterward:

```suggestion
    type_names = {
        TypeError: 'type_error',
        AssertionError: 'assertion_error'
    }
    base_name = type_names.get(cls, 'value_error')
```

Reply:
I'd also add this above the check:
```python
    if cls in (TypeError, ValueError):
        # just TypeError or ValueError, no extra code
        return base_name
```
and modify that check to have AssertionError in the list (so assertion_error doesn't get a special code)

Reply:
What about the `issubclass` check, though?
https://github.com/samuelcolvin/pydantic/blob/18d4b2bb2ac9790bd58780dba67947a088725731/pydantic/error_wrappers.py#L109

Reply:
D’oh, that’s a good point. The version you included above seems good to me.

I also think it could be reasonable to just do a separate check for AssertionError first using essentially a guard pattern — if it’s an AssertionError (or a subclass I suppose, though that should probably be avoided in practice) just return ‘assertion_error’ without looking for a code. Otherwise do what it used to do.

(The code is used by more clearly typed errors typically coming from built-in validation, but I would view the use of an AssertionError as explicitly choosing to ignore such nuance. So I think the code can be ignored for that case.)